src/lib.rs

```diff
@@ -66,15 +68,50 @@ use {
   prometheus_metrics::serve_prometheus,
 };
 
+#[derive(Parser, Debug)]
+#[command(
+  version,
+  about = "A link aggregator for the fediverse",
+  long_about = "A link aggregator for the fediverse.\n\nThis is the Lemmy backend API server. This will connect to a PostgreSQL database, run any pending migrations and start accepting API requests."
+)]
+pub struct CmdArgs {
```

Comment:
You should also make all the fields `pub` to ensure that `CmdArgs` can be constructed outside Lemmy.

Reply:
that seems like kind of a separate decision whether to actually make this part of the public interface

Reply:
Well it already is part of the public interface as it is given to the `start_lemmy_server` function. How should someone who wants to embed lemmy call `start_lemmy_server` if they can't construct the `CmdArgs`? (and they don't want to parse them from command line args obviously).

Reply:
maybe it should also be called `LemmyArgs` as well, since they may not actually come from the command line.

Reply:
is starting it as a library actually supported/ documented anywhere?

Reply:
It was done in this one https://github.com/LemmyNet/lemmy/pull/2618

And of course it is documented on docs.rs, https://docs.rs/lemmy_server/latest/lemmy_server/

Reply:
Agree with this. Lemmybb can run Lemmy as embedded library, although that project isnt maintained now because I dont have time. Anyway I dont see any downside to making them pub.